aws_helper/config.go

```diff
@@ -56,9 +56,14 @@ func CreateAwsSessionFromConfig(config *AwsSessionConfig, terragruntOptions *opt
 		DisableComputeChecksums: aws.Bool(config.DisableComputeChecksums),
 	}
 
+	profile := config.Profile
+	if profile == "" && terragruntOptions.IAMRoleOptions.RoleARN != "" {
+		profile = terragruntOptions.IAMRoleOptions.Profile
+	}
+
 	var sessionOptions = session.Options{
 		Config:            awsConfig,
-		Profile:           config.Profile,
+		Profile:           profile,
```

Comment:
We should give precedence to the config's settings, as we do below for other `terragruntOptions.IAMRoleOptions`.